### ISSUE TITLE
update of shipping choice.

### ISSUE THREAD

**Issue by proseLA** [lat9/one_page_checkout#305]
when one updates their shipping choice there is an ajax call:

ajax.php?act=ajaxOnePageCheckout&method=updateShipping

i am not sure why this call is needed.

i'm pretty sure all of this could be done on the client side using javascript.  all of the amounts are there.

am i wrong?

**Comment by lat9** [comment]
Because, for instance, the customer could have chosen a UPS Next Day delivery and gotten distracted so that the shipping method is no longer available.

**Comment by proseLA** [comment]
i would like to revisit this issue, as i truly feel we are penalizing the many for the advantage of the few.

how would you feel about setting a timer on the resubmit?  it just seems we have all of the information already, and if a customer changes their shipping method 15 seconds after getting to the page, making an ajax call in the majority of cases seems superfluous.

we could add the following code at the top of the javascript (after this section):
https://github.com/lat9/one_page_checkout/blob/071536c741a7e5def8e3d11d36e23344d3b222ef/includes/modules/pages/checkout_one/jquery.checkout_one.js#L1-L8

```
let resubmit = false;

setTimeout(function(){
    resubmit = true;
}, 20000);
```
and then add a conditional for the ajax call.

if there is no need for the ajax call, we could use javascript to recalculate all the totals on the page without the expensive call to the server and then out to all of the carriers.

in this example they would have 20 seconds (maybe it should be 10 minutes?  or potentially configurable by the store owner?)

thoughts?
